### PR TITLE
introduce ExecuteAfmService

### LINKED ISSUE #570: Add missing ExecuteAfmService
We have all the dtos, but service is missing, add one.

### PR DESCRIPTION
**Pull request by jimirocks**
resolves #570

### CODE REVIEW

File: src/main/java/com/gooddata/executeafm/ExecuteAfmService.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import com.gooddata.FutureResult;
+import com.gooddata.GoodDataException;
+import com.gooddata.GoodDataRestException;
+import com.gooddata.GoodDataSettings;
+import com.gooddata.PollResult;
+import com.gooddata.SimplePollHandler;
+import com.gooddata.executeafm.response.ExecutionResponse;
+import com.gooddata.executeafm.result.ExecutionResult;
+import com.gooddata.project.Project;
+import org.springframework.http.HttpStatus;
```
Comment (liry) [inline src/main/java/com/gooddata/executeafm/ExecuteAfmService.java]: needless import

File: src/main/java/com/gooddata/executeafm/ExecuteAfmService.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+     * @param execution execution
+     * @return response of the submitted execution
+     */
+    public ExecutionResponse execute(final Project project, final Execution execution) {
+        try {
+            return restTemplate.postForObject(
+                    EXECUTION_URI,
+                    notNull(execution, "execution"),
+                    ExecutionResponse.class,
+                    notNull(project, "project").getId());
```
Comment (liry) [inline src/main/java/com/gooddata/executeafm/ExecuteAfmService.java]: what if project.getId is null?

File: src/main/java/com/gooddata/executeafm/ExecuteAfmService.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        super(restTemplate, settings);
+    }
+
+    /**
+     * Executes the given execution returning the execution response
+     * @param project project of the execution
+     * @param execution execution
+     * @return response of the submitted execution
+     */
+    public ExecutionResponse execute(final Project project, final Execution execution) {
```
Comment (liry) [inline src/main/java/com/gooddata/executeafm/ExecuteAfmService.java]: Is null response valid state? If not, plz throw ex. if null returned from backend

File: src/main/java/com/gooddata/executeafm/ExecuteAfmService.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        }
+    }
+
+    /**
+     * Get for result of given response.
+     * @param executionResponse response to get the result
+     * @return future of execution result
+     */
+    public FutureResult<ExecutionResult> getResult(final ExecutionResponse executionResponse) {
+        return new PollResult<>(this, new ExecutionPollHandler(notNull(executionResponse, "executionResponse").getExecutionResultUri()));
```
Comment (liry) [inline src/main/java/com/gooddata/executeafm/ExecuteAfmService.java]: what if getExecutionResultUri return null?

File: src/test/groovy/com/gooddata/executeafm/ResultPageTest.groovy [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        'empty limits'   | [1]     | []
+        'different size' | [1, 2]  | [1]
+    }
+
+    def "should get query params"() {
+        when:
+        def page = new ResultPage([1, 2], [7, 8])
+
+        then:
+        page?.offsetsQueryParam == '1%2C2'
```
Comment (liry) [inline src/test/groovy/com/gooddata/executeafm/ResultPageTest.groovy]: null safe not necessary here

File: src/test/java/com/gooddata/executeafm/ExecuteAfmServiceAT.java [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+import com.gooddata.executeafm.response.MeasureGroupHeader;
+import com.gooddata.executeafm.response.ResultDimension;
+import com.gooddata.executeafm.result.DataList;
+import com.gooddata.executeafm.result.DataValue;
+import com.gooddata.executeafm.result.ExecutionResult;
+import com.gooddata.executeafm.result.ResultHeaderItem;
+import org.testng.annotations.Test;
+
+import java.util.Collection;
+import java.util.Collections;
```
Comment (liry) [inline src/test/java/com/gooddata/executeafm/ExecuteAfmServiceAT.java]: unused import

File: src/main/java/com/gooddata/executeafm/ExecuteAfmService.java (hunk 4 above)
Comment (jimirocks) [reply src/main/java/com/gooddata/executeafm/ExecuteAfmService.java]: checked in getter

**Pull request merged**